54370 site icon post review

kebbet opened the pull request:
https://core.trac.wordpress.org/ticket/54370

Work In Progess. Feel free to help out.

kebbet committed:
Fix typo

kebbet committed:
Adds `data-choose` attribute for media modal

kebbet committed:
Adds docblocks, removes wrapping function.

kebbet committed:
Adds since tag, follow coding standards I guess.

kebbet committed:
Store selected elements in vars.

kebbet committed:
Updates docblock, move `frame` into initializing function

kebbet committed:
Reduce calls to .attr(). Remove `classes` var

afercia commented on src/js/_enqueues/admin/site-icon.js:
```diff
@@ -1,138 +1,180 @@
+ * @param {string} url The URL of the site icon image.
+ */
+function switchToUpdate( url ) {
+	var chooseLink = jQuery( '#choose-from-library-link' );
```
I'd consider to rename chooseLink and its selector (also on the PHP side) because it's not a link. It's a button.

afercia commented on src/wp-admin/css/forms.css:
```diff
@@ -817,7 +817,7 @@ ul#add-to-blog-users {
 }
 
-.button-ad-site-icon:focus{
+.button-add-site-icon:focus{
```
Missing space before the opening bracket (it applies to all the new CSS added in this file as well).

afercia commented on src/wp-admin/options-general.php:
```diff
@@ -139,6 +139,7 @@
 			data-alt-classes="<?php echo esc_attr( $classes_for_button_on_change ); ?>"
 			data-size="512"
+			data-choose="<?php esc_attr_e( 'Choose a Site Icon' ); ?>"
 			data-choose-text="<?php esc_attr_e( 'Choose a Site Icon' ); ?>"
```
Why not reusing the existing 'data-choose-text' instead of adding the same string in 'data-choose'?

afercia commented on src/js/_enqueues/admin/site-icon.js:
```diff
@@ -1,138 +1,180 @@
+			},
+			states: [
+				new wp.media.controller.Library({
+					title: $el.data( 'choose' ),
```
This could be changed to 'choose-text' to avoid to add an unnecessary string on the PHP side.

afercia commented on src/js/_enqueues/admin/site-icon.js:
```diff
@@ -1,138 +1,180 @@
+ * @param {Object} attachment The attachment object representing the image.
+ * @return {Object} The image selection options.
+ */
+function calculateImageSelectOptions( attachment ) {
```
I'm not sure moving `calculateImageSelectOptions` and `switchToUpdate` out from the Immediately Invoked Function Expression (IIFE) is recommended. An IIFE also serves the purpose of 'encapsulating' other functions and variables in the IIFE function scope. Instead, to my understanding, these functions would be now in the global scope, which is not recommended. I think most of the WordPress JS files use an IIFE  for this reason: avoid global scope pollution and potential names collision.

kebbet replied:
I guess this is resolved with [edb1546](https://github.com/WordPress/wordpress-develop/pull/6109/commits/edb154606304840d57181c6c9c965eee68541dc1)

afercia commented on src/js/_enqueues/admin/site-icon.js:
```diff
@@ -1,138 +1,180 @@
-(function($) {
+/* global wp, jQuery */
```
I previously reported that to add the comment for globals because I seem to recall the lack of uaed to make jshint complain. I may be wrong though, if that's the case I stand corrected. These gloabls are already registered in thte `.jshintrc` file. See https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/#globals

aaronjorbin replied:
I would keep these even if no longer necessary for `jshint` since every neighbor file has them. Additionally, it does make it easier to understand what the file relies on.

kebbet replied:
The comment on globals is now back in top of the file.

kebbet committed:
Adds space before the opening brackets

kebbet committed:
Reuse existing attribute, rename selector (in PHP)

kebbet committed:
Address review comments from @afercia

kebbet committed:
Merge branch 'WordPress:trunk' into 54370-site-icon-post-review

mukeshpanchal27 commented on src/js/_enqueues/admin/site-icon.js:
```diff
@@ -1,138 +1,178 @@
-			$(img).attr('src', url );
+		frame.open();
+	});
+}( jQuery ));
```
```suggestion
})( jQuery );
```

mukeshpanchal27 commented on src/js/_enqueues/admin/site-icon.js:
```diff
@@ -1,138 +1,178 @@
+		chooseButton.text( chooseButton.attr( 'data-choose-text' ) );
 	});
-}(jQuery));
+}( jQuery ));
```
```suggestion
})( jQuery );
```

kebbet committed:
JS-fixes

kebbet committed:
Focus loss and fixed button ID for triggering modal.

sofialeena commented:
Good point, that makes sense.
{{{
#!html
<a href="https://tassel.pk"></a>
<a href="https://aklasbelakarachi.com"></a>
<a href="https://dailylasbelagamekarachi.com"></a>
<a href="https://cheapcamshows.com"></a>
<a href="https://thebritaintimes.co.uk"></a>
<a href="https://expertzcreation.com"></a>
}}}

afercia commented on src/js/_enqueues/admin/site-icon.js:
```diff
@@ -1,138 +1,180 @@
+	chooseButton.text(
+		chooseButton.attr( 'data-update-text' )
+	);
+	chooseButton.trigger( 'focus' );
```
This should not be necessary because the media modal dialog already sets focus back to the button.

afercia replied:
```suggestion
```

kebbet committed:
Removes unnecessary setting of focus

afercia commented on src/js/_enqueues/admin/site-icon.js:
```diff
@@ -1,138 +1,180 @@
-		$( '#choose-from-library-link' ).text( $( '#choose-from-library-link' ).attr( 'data-choose-text' ) );
-		// Set the state of the button so it can be changed on new icon.
-		$( '#choose-from-library-link' ).attr( 'data-state', '');
+		chooseButton.text( chooseButton.attr( 'data-choose-text' ) );
+		chooseButton.trigger( 'focus' );
```
```suggestion
		chooseButton
			.text( chooseButton.attr( 'data-choose-text' ) )
			.trigger( 'focus' );
```

afercia replied:
Good fix thanks. Moving focus to the choose button seeems the most appropriate thing to do 👍🏽

afercia commented on src/js/_enqueues/admin/site-icon.js:
```diff
@@ -1,138 +1,180 @@
+	var frame;
+	// Build the choose from library frame.
+	$( '#choose-from-library-button' ).on( 'click', function() {
+		var $el = $(this);
```
```suggestion
		var $el = $( this );
```

afercia commented on src/js/_enqueues/admin/site-icon.js:
```diff
@@ -1,138 +1,180 @@
+		});
+
+		frame.on( 'cropped', function( attachment ) {
+			$( '#site_icon_hidden_field' ).val(attachment.id);
```
```suggestion
			$( '#site_icon_hidden_field' ).val( attachment.id );
```

afercia commented on src/js/_enqueues/admin/site-icon.js:
```diff
@@ -1,138 +1,180 @@
+
+		frame.on( 'cropped', function( attachment ) {
+			$( '#site_icon_hidden_field' ).val(attachment.id);
+			switchToUpdate(attachment.url);
```
```suggestion
			switchToUpdate( attachment.url );
```

afercia commented on src/js/_enqueues/admin/site-icon.js:
```diff
@@ -1,138 +1,180 @@
+				$el.data( 'size' ) === attachment.attributes.width
+			) {
+				// Set the value of the hidden input to the attachment id.
 				$( '#site_icon_hidden_field' ).val(attachment.id);
```
```suggestion
				$( '#site_icon_hidden_field' ).val( attachment.id );
```

afercia commented on src/js/_enqueues/admin/site-icon.js:
```diff
@@ -1,138 +1,180 @@
+				// Set the value of the hidden input to the attachment id.
 				$( '#site_icon_hidden_field' ).val(attachment.id);
-				switchToUpdate(attachment.url);
+				switchToUpdate(attachment.attributes.url);
```
```suggestion
				switchToUpdate( attachment.attributes.url );
```

afercia commented on src/js/_enqueues/admin/site-icon.js:
```diff
@@ -1,138 +1,180 @@
+	// Set site-icon-img src to the url and remove the hidden class.
+	iconPreview.find( 'img' ).not( '.browser-preview' )
+		.each( function( i, img ) {
+			jQuery(img).attr( 'src', url );
```
```suggestion
			jQuery( img ).attr( 'src', url );
```

afercia commented on src/js/_enqueues/admin/site-icon.js:
```diff
@@ -1,138 +1,180 @@
 		$( '#site_icon_hidden_field' ).val( 'false' );
 		$( '#site-icon-preview' ).toggleClass( 'hidden' );
-		$( this ).toggleClass( 'hidden' );
+		$(this).toggleClass( 'hidden' );
```
```suggestion
		$( this ).toggleClass( 'hidden' );
```

afercia commented on src/wp-admin/css/forms.css:
```diff
@@ -801,23 +801,23 @@ ul#add-to-blog-users {
 .button-add-site-icon:focus,
-.button-add-site-icon:hover{
+.button-add-site-icon:hover {
 	background: white;
```
```suggestion
	background: #fff;
```

afercia replied:
WordPress doesn't use named colors

kebbet committed:
Coding standard fixes.

aaronjorbin commented on src/js/_enqueues/admin/site-icon.js:
```diff
@@ -1,138 +1,180 @@
+			'data-state': ''
+		});
 
 		// Swap the text of the button.
```
If it's no longer only swapping the text, this comment needs to be updated

aaronjorbin commented on src/js/_enqueues/admin/site-icon.js:
```diff
@@ -1,138 +1,180 @@
+ * @param {jQuery} $ The jQuery object.
+ */
+( function ( $ ) {
+	var chooseButton = $( '#choose-from-library-button' );
```
as a jQuery object, the variable name should start with `$`

aaronjorbin commented on src/js/_enqueues/admin/site-icon.js:
```diff
@@ -1,138 +1,180 @@
-		var classes = $( '#choose-from-library-link' ).attr( 'class' );
-		$( '#choose-from-library-link' ).attr( 'class', $( '#choose-from-library-link' ).attr( 'data-alt-classes' ) );
-		$( '#choose-from-library-link' ).attr( 'data-alt-classes', classes );
+		chooseButton.attr({
```
I am finding this harder to groc than the explicit calls to each.  If you want to go this route, I think it should get a comment describing what is going on

kebbet replied:
Added a three liner comment on it. Better?

aaronjorbin commented on src/js/_enqueues/admin/site-icon.js:
```diff
@@ -1,138 +1,180 @@
+ */
+function switchToUpdate( url ) {
+	var chooseButton = jQuery( '#choose-from-library-button' );
+	var iconPreview = jQuery( '#site-icon-preview' );
```
Var statements should be combined.

aaronjorbin commented on src/js/_enqueues/admin/site-icon.js:
```diff
@@ -1,138 +1,180 @@
+})( jQuery );
+
+/**
+ * Updates the site icon preview with the specified URL.
```
This skips some of the changes that are taking place. Mabe a short description along the lines of "Updates Site Icon settings after a selection is made" and then this along with an overview of the other changes?

aaronjorbin reviewed (changes requested):
Thanks @kebbet!  I've left a handful of specifics, but in general:

1. Keeping everything in the IIFE since nothing from site-icon.js should be in the global namespace nor considered an API with any promise of backward compatibility. Additionally, it means `$` can be used everywhere instead of `jQuery`. 
2. Variable names for jQuery objects should start with `$` so that it's easier to tell at a glance that these are jQuery objects.
3. Make sure the docblocks are accurate everywhere. 
4. There should be a blank line above inline comments.

kebbet committed:
Merge branch 'WordPress:trunk' into 54370-site-icon-post-review

kebbet committed:
Updates docblocks, combine var-statements, start jQuery objects with $. Wrap feature in a function

kebbet committed:
Merge branch 'WordPress:trunk' into 54370-site-icon-post-review

kebbet committed:
Accessibility: adds better alternative text to the images

kebbet committed:
CS-issue with to many tabs.

kebbet committed:
CS-issue

kebbet committed:
Missing semicolons.

kebbet committed:
Handle alt-attribute on remove

kebbet committed:
Remove debug log.

kebbet commented:
I have tested the PHP-part on a multisite install, and correct alternative text is selected for the preview images.

afercia commented on src/js/_enqueues/admin/site-icon.js:
```diff
@@ -41,22 +58,31 @@
+	 *
+	 * @since 6.5.0
+	 */
+	( function() {
```
Maybe I'm missing something but I'm not sure this IIFE is needed? All the `on()` functions within this block of code are already wrapped within the main IIFE.

kebbet replied:
I removed it.

afercia commented on src/js/_enqueues/admin/site-icon.js:
```diff
@@ -99,40 +130,98 @@
+
+			// Set site-icon-img src to the url and remove the hidden class.
+			$iconPreview.find( 'img' ).not( '.browser-preview' )
+			.each( function( i, img ) {
+				$( img )
+				.attr({
+					'src': attributes.url,
+					'alt': i18nAltString 
+				});
+			});
```
I think these lines should be indented by one tab.

afercia commented on src/js/_enqueues/admin/site-icon.js:
```diff
@@ -99,40 +130,98 @@
+	 *
+	 * @since 6.5.0
+	 */
+	( function () {
```
Same as above: I'm not sure this IIFE is really needed?

afercia commented on src/js/_enqueues/admin/site-icon.js:
```diff
@@ -99,40 +130,98 @@
+			$( this ).toggleClass( 'hidden' );
+			$iconPreview.toggleClass( 'hidden' );
+			$iconPreview.find( 'img' ).not( '.browser-preview' )
+			.each( function( i, img ) {
+				$( img )
+				.attr({
+					'src': '',
+					'alt': '' 
+				});
+			});
```
I think these lines should be indented by one tab.

kebbet replied:
And maybe indent .attr-lines as well one more line? What do you think @afercia ?

Like this:
```
$iconPreview.find( 'img' ).not( '.browser-preview' )
	.each( function( i, img ) {
		$( img )
			.attr({
				'src': '',
				'alt': '' 
			});
	});
```

afercia replied:
Oh yes, I missed that 👍🏽

afercia commented:
Thanks for working on crafting a more meaningful alt text. The alt text value looks good to me and it's consistent with the pattern already in use for the media widgets and the Gutenberg featured image. To recap:

When the image does have an alt text:
`Current image: %s` where ` %s` is the alt text.
Otherwise:
`The current image has no alternative text. The file name is: %s` where ` %s` is the image filename.

However, if we want to keep the current design, there's one more thing to address. While the featured image is only _one_ image, the site icon UI uses _two_ images: one for previewing the site icon as it is used by most browsers in the browser tab and another one for previewing the site icon as it is used by mobile devices, as an 'app icon'. 


I think it would be good to consider to:
- Either distinguish the two image, by prefixing something like 'Browser icon preview' and 'App icon preview'
- Or provide the alt text only once, by describing both previews only once (technical details to be decided)

@alexstine I'd appreciate your feedback here, keeping in mind that sighted screen reader users do see two images.

kebbet committed:
Remove IIFE-sections, adjusts indents. Move var to function scope

kebbet committed:
Merge branch 'WordPress:trunk' into 54370-site-icon-post-review

kebbet committed:
Temp reset

kebbet committed:
Merge branch '54370-site-icon-post-review' of https://github.com/kebbet/wordpress-develop into 54370-site-icon-post-review

kebbet committed:
Remove IIFE-sections, adjusts indents. Move var to function scope

alexstine commented:
@afercia 

> Either distinguish the two image, by prefixing something like 'Browser icon preview' and 'App icon preview'

I like this approach.

kebbet committed:
Merge branch 'WordPress:trunk' into 54370-site-icon-post-review

kebbet committed:
Adds prefix to alt text for each preview image.

afercia commented on src/wp-admin/options-general.php:
```diff
@@ -120,8 +120,11 @@
+	$img_alt_value           = '';
+	$site_icon_id            = (int) get_option( 'site_icon' );
+	$app_icon_alt_prefix     = _x( 'App icon preview', 'Prefix for alternative text of the site icon preview' );
+	$browser_icon_alt_prefix = _x( 'Browser icon preview', 'Prefix for alternative text of the site icon preview' );
```
Ideally, the context parameter of `_x()` should only contain very short text to clarify the context e.g. verb, name, or the object the string refers to e.g. ctegories, tags, etc. See usages in core. The actual usage of this parameter in Gutenberg is sometimes not ideal, please refer to core instead.

afercia commented on src/js/_enqueues/admin/site-icon.js:
```diff
@@ -141,32 +143,52 @@
+
+		i18nAppAlternativeString = wp.i18n.sprintf(
+			/* Translators: 1: Prefix used for alternative text of the preview of the site icon. 2: Alternative text of the site icon image, */
+			wp.i18n.__( '%1$s: %2$s' ),
```
This would solve the a11y need to clarify the alt text. However, I'm not sure concatenation is ideal from an l10n perspective and for translators. Ideally, these should be full sentences, avoiding concatenation.

aaronjorbin replied:
It's also best to use the same strings as are used in PHP so that they are translated the same. Additionally, while in English (and some other languages), it makes sense to put the alt text after the prefix, this isn't something that can be assumed for all languages (Some could move the `$s` elsewhere in the sentence).

kebbet replied:
Placeholders could be shuffled during translations, i.e. `%$2s (%1$s)`, thats why I created at translatable concatenation  string, however, I added four full strings that can be translated independently now.

afercia commented on src/js/_enqueues/admin/site-icon.js:
```diff
@@ -141,32 +143,52 @@
+
+		i18nBrowserAlternativeString = wp.i18n.sprintf(
+			/* Translators: 1: Prefix used for alternative text of the preview of the site icon. 2: Alternative text of the site icon image, */
+			wp.i18n.__( '%1$s: %2$s' ),
```
Se comment above re: concatenation

aaronjorbin commented on src/js/_enqueues/admin/site-icon.js:
```diff
@@ -1,15 +1,40 @@
+/* global jQuery, wp */
+
+( function( $ ) {
+	var $chooseButton = $( '#choose-from-library-button' ),
+	$iconPreview = $( '#site-icon-preview' ),
+	$browserIconPreview = $( '#browser-icon-preview' ),
+	$appIconPreview = $( '#app-icon-preview' ),
+	$hiddenDataField = $( '#site_icon_hidden_field' ),
+	$removeButton = $( '#js-remove-site-icon' ),
+	frame;
```
```suggestion
	var $chooseButton = $( '#choose-from-library-button' ),
		$iconPreview = $( '#site-icon-preview' ),
		$browserIconPreview = $( '#browser-icon-preview' ),
		$appIconPreview = $( '#app-icon-preview' ),
		$hiddenDataField = $( '#site_icon_hidden_field' ),
		$removeButton = $( '#js-remove-site-icon' ),
		frame;
```

Variables after the first should be indented.

https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/#declaring-variables-with-var

aaronjorbin commented on src/js/_enqueues/admin/site-icon.js:
```diff
@@ -1,15 +1,40 @@
-			x1, y1, imgSelectOptions;
+			xImg = xInit,
+			yImg = yInit,
+			x1,
+			y1,
+			imgSelectOptions;
```
```suggestion
			x1, y1, imgSelectOptions;
```

declarations can be grouped on a single line

https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/#declaring-variables-with-var

aaronjorbin commented on src/js/_enqueues/admin/site-icon.js:
```diff
@@ -41,98 +66,178 @@
+	 * @param {array} attributes The attributes for the attachment.
+	 */
+	function switchToUpdate( attributes ) {
+		var i18nAltString,
+		i18nAppAlternativeString,
+		i18nBrowserAlternativeString;
```
```suggestion
		var i18nAltString, i18nAppAlternativeString, i18nBrowserAlternativeString;
```

Declerations can be grouped. 

https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/#declaring-variables-with-var

aaronjorbin commented:
> However, if we want to keep the current design, there's one more thing to address. While the featured image is only one image, the site icon UI uses two images: one for previewing the site icon as it is used by most browsers in the browser tab and another one for previewing the site icon as it is used by mobile devices, as an 'app icon'.

See https://core.trac.wordpress.org/ticket/54370#comment:86 for a codepen with the updated design. While there are visual updates, this suggestion still stands since there are two images.

kebbet committed:
Address feedback from @afercia

kebbet committed:
CS-fixes

kebbet committed:
CS-fixes

kebbet committed:
Update src/js/_enqueues/admin/site-icon.js

Co-authored-by: Aaron Jorbin <aaronjorbin@users.noreply.github.com>

kebbet committed:
Address feedback from @aaronjorbin

kebbet committed:
fix

kebbet commented:
If someone wants to continue the work on this and update the feature with the suggested new design, feel free to take over, this PR now fixes whats currently in trunk.

aaronjorbin commented:
@kebbet Thank you!  I'll handle getting the updated design wired in!

aaronjorbin committed:
Merge branch 'trunk' of github.com:WordPress/wordpress-develop into 54370-site-icon-post-review

aaronjorbin committed:
Start wiring in new UI

aaronjorbin committed:
Update text for description

aaronjorbin committed:
Ensure unused dependecies aren't enqueued and translations are properly set

aaronjorbin committed:
Set IDs so JS works

aaronjorbin committed:
Minor tweaks to comments

aaronjorbin committed:
Ensure icons can be hidden

aaronjorbin committed:
use less generic class names

sofialeena commented:
Feel free to take over and implement the suggested new design. This PR fixes issues currently present in trunk.
{{{
#!html
<a href="https://1keonhacai.net"></a><a href="https://nhandinhbongda.bet"></a><a href="https://cheaplivegirls.com"></a><a href="https://businessjem.com"></a>
}}}

kebbet commented:
It is no longer possible to crop the image. Why was the cropping dependency remove here [80fa6d0](https://github.com/WordPress/wordpress-develop/pull/6109/commits/80fa6d0e354b05f486b91f9c9f7ccbc3393f81ad)?

kebbet commented on src/wp-admin/css/site-icon.css:
```diff
@@ -75,3 +75,111 @@
+	margin-bottom: 20px;
+}
+
+  .site-icon-preview.settings .home-icon {
+	width: 58px;
+	height: 58px;
+	border-radius: 12.95px;
+  }
```
```suggestion
.site-icon-preview.settings .home-icon {
	width: 58px;
	height: 58px;
	border-radius: 12.95px;
}
```

aaronjorbin replied:
Fixed this one manually since I needed to change the selector to fix cropping

kebbet commented on src/wp-admin/css/site-icon.css:
```diff
@@ -75,3 +75,111 @@
+	gap: 16px;
+	flex: 1 0 0;
+	z-index: 1;
+	border-radius: 9.824px 0px 0px 0px;
```
```suggestion
	border-radius: 9.824px 0 0 0;
```

kebbet commented on src/wp-admin/css/site-icon.css:
```diff
@@ -75,3 +75,111 @@
+	border-top: 1px solid rgba(255, 255, 255, 0.2);
+	border-left: 1px solid rgba(255, 255, 255, 0.2);
+	background: linear-gradient(180deg, #DCDCDE 0%, #BDBDBD 100%);
+	box-shadow: 0px 10px 22px 0px rgba(0, 0, 0, 0.2), 0px 40px 40px 0px rgba(0, 0, 0, 0.17), 0px 89px 53px 0px rgba(0, 0, 0, 0.1), 0px 158px 63px 0px rgba(0, 0, 0, 0.03), 0px 247px 69px 0px rgba(0, 0, 0, 0);
```
```suggestion
	box-shadow: 0 10px 22px 0 rgba(0, 0, 0, 0.2), 0 40px 40px 0 rgba(0, 0, 0, 0.17), 0 89px 53px 0 rgba(0, 0, 0, 0.1), 0 158px 63px 0 rgba(0, 0, 0, 0.03), 0 247px 69px 0 rgba(0, 0, 0, 0);
```

kebbet commented on src/wp-admin/css/site-icon.css:
```diff
@@ -75,3 +75,111 @@
+	border-radius: 9.824px 0px 0px 0px;
+	border-top: 1px solid rgba(255, 255, 255, 0.2);
+	border-left: 1px solid rgba(255, 255, 255, 0.2);
+	background: linear-gradient(180deg, #DCDCDE 0%, #BDBDBD 100%);
```
```suggestion
	background: linear-gradient(180deg, #dcdcde 0%, #bdbdbd 100%);
```

kebbet commented on src/wp-admin/css/site-icon.css:
```diff
@@ -75,3 +75,111 @@
+}
+
+.site-icon-preview-browser-buttons {
+	fill: #8C8F94;
```
```suggestion
	fill: #8c8f94;
```

kebbet commented on src/wp-admin/css/site-icon.css:
```diff
@@ -75,3 +75,111 @@
+	gap: 8px;
+	flex: 1 0 0;
+	border-radius: 4px;
+	background: #F6F7F7;
```
```suggestion
	background: #f6f7f7;
```

kebbet commented on src/wp-admin/css/site-icon.css:
```diff
@@ -75,3 +75,111 @@
+	flex: 1 0 0;
+	border-radius: 4px;
+	background: #F6F7F7;
+	box-shadow: 0px 1px 3px 0px rgba(0, 0, 0, 0.04), 0px 5px 5px 0px rgba(0, 0, 0, 0.03), 0px 12px 7px 0px rgba(0, 0, 0, 0.02), 0px 22px 9px 0px rgba(0, 0, 0, 0.01), 0px 34px 9px 0px rgba(0, 0, 0, 0);
```
```suggestion
	box-shadow: 0 1px 3px 0 rgba(0, 0, 0, 0.04), 0 5px 5px 0 rgba(0, 0, 0, 0.03), 0 12px 7px 0 rgba(0, 0, 0, 0.02), 0 22px 9px 0 rgba(0, 0, 0, 0.01), 0 34px 9px 0 rgba(0, 0, 0, 0);
```

kebbet commented on src/wp-admin/css/site-icon.css:
```diff
@@ -75,3 +75,111 @@
+	flex-shrink: 0;
+	z-index: 1;
+	border-radius: 9.824px;
+	box-shadow: 0px 1.517px 3.034px 0px rgba(0, 0, 0, 0.1), 0px 4.552px 4.552px 0px rgba(0, 0, 0, 0.09), 0px 10.621px 6.828px 0px rgba(0, 0, 0, 0.05), 0px 19.724px 7.586px 0px rgba(0, 0, 0, 0.01), 0px 30.345px 8.345px 0px rgba(0, 0, 0, 0);
```
```suggestion
	box-shadow: 0 1.517px 3.034px 0 rgba(0, 0, 0, 0.1), 0 4.552px 4.552px 0 rgba(0, 0, 0, 0.09), 0 10.621px 6.828px 0 rgba(0, 0, 0, 0.05), 0 19.724px 7.586px 0 rgba(0, 0, 0, 0.01), 0 30.345px 8.345px 0 rgba(0, 0, 0, 0);
```

kebbet commented on src/wp-admin/css/site-icon.css:
```diff
@@ -75,3 +75,111 @@
+
+.site-icon-preview.settings {
+	width: 350px;
+	padding: 16px 0px 0px 16px;
```
```suggestion
	padding: 16px 0 0 16px;
```

kebbet commented on src/wp-admin/css/site-icon.css:
```diff
@@ -75,3 +75,111 @@
+.site-icon-preview {
+	display: flex;
+	height: 60px;
+	padding: 8px 0px 0px 8px;
```
```suggestion
	padding: 8px 0 0 8px;
```

kebbet commented on src/wp-admin/css/site-icon.css:
```diff
@@ -75,3 +75,111 @@
+}
+
+.site-icon-preview-close-button {
+	fill: #50575E;
```
```suggestion
	fill: #50575e;
```

kebbet commented:
@aaronjorbin CSS colors should be all lowercase, 0 should be unitless, see https://developer.wordpress.org/coding-standards/wordpress-coding-standards/css/#properties

afercia commented:
> It is no longer possible to crop the image. Why was the cropping dependency remove here [80fa6d0](https://github.com/WordPress/wordpress-develop/pull/6109/commits/80fa6d0e354b05f486b91f9c9f7ccbc3393f81ad)?

Latest commits broke the crop step for me. This is what I get in the media modal dialog in the 'crop' step: missing crop UI and the site icon preview is displayed in the sidebar.... 😱  No JS errors.

![Screenshot 2024-02-21 at 09 36 45](https://github.com/WordPress/wordpress-develop/assets/1682452/b935fa0e-f7f4-4647-8358-b1396832e81a)

Edit:
> the site icon preview is displayed in the sidebar

Ok I realize the sidebar is supposed to show a preview but regardless the latest CSS changes completely broke the media modal crop frame. The selected image isn't visible, The crop handlers are misplaced and appear at the top left of the modal dialog content,as such cropping is impossible.

afercia commented on src/wp-admin/css/site-icon.css:
```diff
@@ -75,3 +75,107 @@
+}
+
+.site-icon {
+	width: 24px;
```
this width breaks the displaying of the image in the media modal

afercia commented:
In `src/wp-admin/css/site-icon.css`, the CSS rulesets with the following selectors target also elements within the media modal dialog in the crop step, thus breaking the layout:
- .site-icon-preview .favicon-preview
- .site-icon-preview .favicon, .site-icon-preview .browser-title
- .site-icon-preview

This is how the media modal dialog looks like on current trunk:

![trunk](https://github.com/WordPress/wordpress-develop/assets/1682452/8876bb0c-e313-4e5a-b7f9-1c25322d3d1e)


Notice the layout of hte previews in the sidebar is broken.

This is how it is supposed to look, screenshot from WordPress 6.4: notice the previews are separated and vertically stacked:

![6 4](https://github.com/WordPress/wordpress-develop/assets/1682452/fd6c23e6-02ad-4c2b-9bb5-05ad64b82784)

afercia commented:
This pull request was created to address some bugs and other issues after https://core.trac.wordpress.org/changeset/57602
Specifically:

- JS and CSS coding standards 
- Documentation 
- Optimize the jQuery code following best practices for element selections and function calls
- Focus loss when removing the site icon
- Accessibility feedback from https://github.com/WordPress/wordpress-develop/pull/6064#pullrequestreview-1872283972 regarding the alt text.
- Fix the modal frame title
- Fix a typo in a CSS selector that prevented focus style to work as expected

We're now in Beta 2 and new changes were added with the only purpose of updating the design. To me, these additional changes are outside of the scope of this PR. More importantly:
- We're now in Beta 2.
- These additional changes aren't bugs, they are enhancements and should not be committed during Beta 2.
- These changes completely broke the media modal dialog 'crop' step.
- These changes don't follow the coding standards.

I'd recommend to:
- Revert the latest changes.
- Fix the layout of the two previews in the media modal dialog: they should be stacked vertically. This seems a trivial fix as it just needs more specificity for the CSS selectors.
- Review and test thoroughly.
- Commit.

In my opinion, any other change should go in a separate pull request or patch after this PR has been committed, to better evaluate if additional changes can be made during Beta 2.

aaronjorbin committed:
Fix specificity in preview

aaronjorbin commented:
> It is no longer possible to crop the image. Why was the cropping dependency remove here [80fa6d0](https://github.com/WordPress/wordpress-develop/pull/6109/commits/80fa6d0e354b05f486b91f9c9f7ccbc3393f81ad)?

The cropping was broken due to some CSS specification issues that I just fixed. jCrop has long been unused by core and wasn't actually used here (It was a part of the original code from [32994] and I hadn't removed it until now)

aaronjorbin committed:
Use unitless for 0

Co-authored-by: Erik <11491369+kebbet@users.noreply.github.com>

aaronjorbin committed:
Use unitless for 0

Co-authored-by: Erik <11491369+kebbet@users.noreply.github.com>

aaronjorbin committed:
lowercase colors

Co-authored-by: Erik <11491369+kebbet@users.noreply.github.com>

aaronjorbin committed:
lowercase colors

Co-authored-by: Erik <11491369+kebbet@users.noreply.github.com>

aaronjorbin committed:
lowercase colors

Co-authored-by: Erik <11491369+kebbet@users.noreply.github.com>

aaronjorbin committed:
Use unitless for 0

Co-authored-by: Erik <11491369+kebbet@users.noreply.github.com>

aaronjorbin committed:
Use unitless for 0

Co-authored-by: Erik <11491369+kebbet@users.noreply.github.com>

aaronjorbin committed:
lowercase colors

Co-authored-by: Erik <11491369+kebbet@users.noreply.github.com>

aaronjorbin committed:
Use unitless for 0

Co-authored-by: Erik <11491369+kebbet@users.noreply.github.com>

aaronjorbin committed:
Use unitless for 0

Co-authored-by: Erik <11491369+kebbet@users.noreply.github.com>

aaronjorbin committed:
Fix indentation

aaronjorbin commented:
Reverting and moving backward won't help get this in the best shape for the final release. Cropping is now fixed. Going to spend some time today getting things finalized for commit

kebbet commented on src/wp-admin/css/site-icon.css:
```diff
@@ -75,3 +75,111 @@
+.site-icon-preview-inline.settings .home-icon {
+	width: 58px;
+	height: 58px;
+	border-radius: 12.95px;
```
Can all pixel values be changed to be integers?

kebbet commented on src/wp-admin/css/site-icon.css:
```diff
@@ -75,3 +75,111 @@
+	width: 200%;
+	height: 200%;
+	transform: translateX(-25%) translateY(-25%);
+	/* background: linear-gradient(0deg, rgba(0, 0, 0, 0.2) 0%, rgba(0, 0, 0, 0.3) 100%), var( --site-icon-url ) 50%/cover no-repeat; */
```
Remove unused code?

aaronjorbin committed:
Merge branch 'trunk' of github.com:WordPress/wordpress-develop into 54370-site-icon-post-review

kebbet commented on src/wp-admin/css/site-icon.css:
```diff
@@ -75,3 +75,111 @@
+	flex-shrink: 0;
+	z-index: 1;
+	border-radius: 9.824px;
+	box-shadow: 0 1.517px 3.034px 0 rgba(0, 0, 0, 0.1), 0 4.552px 4.552px 0 rgba(0, 0, 0, 0.09), 0 10.621px 6.828px 0 rgba(0, 0, 0, 0.05), 0 19.724px 7.586px 0 rgba(0, 0, 0, 0.01), 0 30.345px 8.345px 0 rgba(0, 0, 0, 0);
```
Can box-shadows be simplifed to use one shadow statement, this is really messy.

kebbet commented on src/wp-admin/css/site-icon.css:
```diff
@@ -75,3 +75,111 @@
+	border-top: 1px solid rgba(255, 255, 255, 0.2);
+	border-left: 1px solid rgba(255, 255, 255, 0.2);
+	background: linear-gradient(180deg, #dcdcde 0%, #bdbdbd 100%);
+	box-shadow: 0 10px 22px 0 rgba(0, 0, 0, 0.2), 0 40px 40px 0 rgba(0, 0, 0, 0.17), 0 89px 53px 0 rgba(0, 0, 0, 0.1), 0 158px 63px 0 rgba(0, 0, 0, 0.03), 0 247px 69px 0 rgba(0, 0, 0, 0);
```
See above.

kebbet commented on src/wp-admin/css/site-icon.css:
```diff
@@ -75,3 +75,111 @@
+	flex: 1 0 0;
+	border-radius: 4px;
+	background: #f6f7f7;
+	box-shadow: 0 1px 3px 0 rgba(0, 0, 0, 0.04), 0 5px 5px 0 rgba(0, 0, 0, 0.03), 0 12px 7px 0 rgba(0, 0, 0, 0.02), 0 22px 9px 0 rgba(0, 0, 0, 0.01), 0 34px 9px 0 rgba(0, 0, 0, 0);
```
See above.

kebbet commented on src/wp-admin/css/site-icon.css:
```diff
@@ -75,3 +75,111 @@
+	border-radius: 12.95px;
+}
+
+.site-icon-preview-inline.settings:after {
```
This background is not updated when a new icon is selected. Use an img-tag or similar instead? And when using solid colors as site icons it is hard to see the icon. Maybe add a opacity to this background, or even skip it. 
And using a black site icon makes the shadow disappear. 

![bild](https://github.com/WordPress/wordpress-develop/assets/11491369/9a9e959c-1bd7-4816-9879-38793d9dcc21)

![bild](https://github.com/WordPress/wordpress-develop/assets/11491369/2d949880-dcb6-490b-92de-f5e3f8086872)


![bild](https://github.com/WordPress/wordpress-develop/assets/11491369/39a4f3a4-5dd5-4a68-a665-7e064ce3c5c1)

![bild](https://github.com/WordPress/wordpress-develop/assets/11491369/d5ec902c-a786-4082-8623-7642e8d5c004)

![bild](https://github.com/WordPress/wordpress-develop/assets/11491369/a50ab40c-da2a-4aa6-89c3-6a10402943d8)

kebbet commented on src/wp-admin/css/site-icon.css:
```diff
@@ -75,3 +75,111 @@
+	height: 44px;
+	flex-shrink: 0;
+	z-index: 1;
+	border-radius: 9.824px;
```
Make it 10px

kebbet commented on src/wp-admin/options-general.php:
```diff
@@ -160,13 +204,10 @@ class="<?php echo esc_attr( $classes_for_button ); ?>"
 		<?php
 			/* translators: %s: Site Icon size in pixels. */
-			printf( __( 'Site Icons should be square and at least %s pixels.' ), '<strong>512 &times; 512</strong>' );
+			printf( __( 'Site Icons are what you see in browser tabs, bookmark bars, and within the WordPress mobile apps. They should be square and at least %s pixels.' ), '<strong>512 &times; 512</strong>' );
```
Maybe wrap size string in a `<code>`-tag instead to align with other descriptive texts on the page?

![bild](https://github.com/WordPress/wordpress-develop/assets/11491369/4c6b564d-8396-46d9-b180-56468fa7ef83)

kebbet reviewed (commented):
Some notes on the current state of the PR.

kebbet commented:
> In `src/wp-admin/css/site-icon.css`, the CSS rulesets with the following selectors target also elements within the media modal dialog in the crop step, thus breaking the layout:
> 
>     * .site-icon-preview .favicon-preview
> 
>     * .site-icon-preview .favicon, .site-icon-preview .browser-title
> 
>     * .site-icon-preview
> 
> 
> This is how the media modal dialog looks like on current trunk:
> 
> ![trunk](https://private-user-images.githubusercontent.com/1682452/306580028-8876bb0c-e313-4e5a-b7f9-1c25322d3d1e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg1Mzk1NTUsIm5iZiI6MTcwODUzOTI1NSwicGF0aCI6Ii8xNjgyNDUyLzMwNjU4MDAyOC04ODc2YmIwYy1lMzEzLTRlNWEtYjdmOS0xYzI1MzIyZDNkMWUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDIyMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAyMjFUMTgxNDE1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9N2Q2NTA1NGRlM2Y4MDhmNWM0MTMwMzIyZmY3YzJjNWNhMjFjZTQzOTQ2YTlmZWJjYjJmM2FjZWY5NDlhYjU5ZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.5ioW7dDx5f8AaocQLLoRPTq3CRvR0Qhy4zdlsXwY4ow)
> 
> Notice the layout of hte previews in the sidebar is broken.
> 
> This is how it is supposed to look, screenshot from WordPress 6.4: notice the previews are separated and vertically stacked:
> 
> ![6 4](https://private-user-images.githubusercontent.com/1682452/306579760-fd6c23e6-02ad-4c2b-9bb5-05ad64b82784.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg1Mzk1NTUsIm5iZiI6MTcwODUzOTI1NSwicGF0aCI6Ii8xNjgyNDUyLzMwNjU3OTc2MC1mZDZjMjNlNi0wMmFkLTRjMmItOWJiNS0wNWFkNjRiODI3ODQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDIyMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAyMjFUMTgxNDE1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZjU1MzEwYTY3NzhkMzIxMzdhNDY5YzU4NjBkNmU1MDhkNWEyNzJmNGRhYWJkM2UyMDIyZTJmNzM4N2Q2MGFkNSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.JM3I-foIfjDy-uj-rXzWuN78DBK38A6PzpO2XBMz3bw)

This needs to be addressed.

kebbet commented:
> Reverting and moving backward won't help get this in the best shape for the final release. Cropping is now fixed. Going to spend some time today getting things finalized for commit

Thanks for clarifying and fixing cropping.

kebbet commented:
First, have a look at the admin bar when the Site Title option is changed in options-general.php. Is it possible to use the same mechanism in the preview of the site icon? Basically update the tab name, while typing in the Site Title field above?

kebbet commented on src/wp-admin/options-general.php:
```diff
@@ -107,34 +107,78 @@
+	<div class="site-icon-preview-browser">
+		<svg role="img" aria-label="" width="48" height="40" fill="none" xmlns="http://www.w3.org/2000/svg" class="site-icon-preview-browser-buttons"><path fill-rule="evenodd" clip-rule="evenodd" d="M0 20a6 6 0 1 1 12 0 6 6 0 0 1-12 0Zm18 0a6 6 0 1 1 12 0 6 6 0 0 1-12 0Zm24-6a6 6 0 1 0 0 12 6 6 0 0 0 0-12Z" /></svg>
+		<div class="site-icon-preview-tab">
+			<img src="<?php site_icon_url(); ?>" class="site-icon" id="browser-icon-preview" />
```
This image tag is now missing the alt text specified in `$browser_icon_alt_value`.

aaronjorbin committed:
Update Title in preview as it chagnes

aaronjorbin committed:
Update background for preview with the new image

aaronjorbin committed:
Remove float which was used in old UI as it is interfering with the preview in the cropper

aaronjorbin committed:
Merge branch 'trunk' of github.com:WordPress/wordpress-develop into 54370-site-icon-post-review

aaronjorbin committed:
Restore alt text for browser and remove svg from screen reader

kebbet committed:
Use <code> tag for size info.

kebbet committed:
Tidy CSS and add border to preview field.

kebbet commented:
With the latest changes from me, this is how the section looks like

![bild](https://github.com/WordPress/wordpress-develop/assets/11491369/0b641c74-7d35-4717-a852-ef4dcfce9922)
![bild](https://github.com/WordPress/wordpress-develop/assets/11491369/46fedb41-370c-41a7-8a09-7fa14e323883)
![bild](https://github.com/WordPress/wordpress-develop/assets/11491369/539e25ef-365b-4969-a54f-c430c91ed08c)
![bild](https://github.com/WordPress/wordpress-develop/assets/11491369/8714a51d-d413-4fd4-a90a-d4caff473066)

RTL
![bild](https://github.com/WordPress/wordpress-develop/assets/11491369/6145b0af-d3e2-4daf-957f-cf6efe625068)

afercia commented:
> Reverting and moving backward won't help get this in the best shape for the final release. 

I kindly disagree. The design changes are outside of the scope of this PR. I'm not sure insisting broadening the scope of this PR is ideal. More importantly, given we're in Beta 2, I'm not sure it's any wise. At this point of the release cycle any change should be as small and self-contained as possible and only focused to fix bugs.

While I agree that updating the design with a more modern browser interface is a nice to have, that's not required for this functionality to work. Also, if updating the design of the preview is desired, I'd like to remind everyone that it would need to be be updated also in other two places; 
- in the media modal dialog sidebar
- in the customizer

I'm not sure providing a different design only in the General Settings page would be ideal and help users as it would be clearly inconsistent and confusing. Extending the design changes to the media modal and customizer in this PR would be really out of scope and make this PR way larger than it was meant to be.

Lastly, @kebbet reported valid concerns about the latest changes that, in my opinion, would need more time to be addressed. As I see it, Beta 2 is not the best timing for such changes. I'd like to make clear that I'm nog against a design update, but that should go in a separate PR so that such changes can be better evaluated to establish if they fit with the release cycle process.

swissspidy commented:
Hi all 👋 

Just chiming in here since I see this ticket / PR has been open for a while now with a lot of activity, so I am just wondering if there's anything I can help with to ensure this is in a good shape to land in time for the next beta.

Could someone perhaps summarize the goals and contents of this pull request for me, maybe with a before/after comparison? Even just updating the PR title & description would be helpful.

Or maybe put another way: what are the issues with the current version in `trunk` that should be addressed in this release?

Thanks in advance 🙏 

-------

In the meantime, since I saw a comment about coding standards, missing whitespace, etc.: Please note that we have tools like Prettier, just use e.g. `npx prettier --write src/js/_enqueues/admin/site-icon.js` to format code according to the WP coding standards. Works with CSS too :)

Also, if the changes in the PR are too big to review in one go, perhaps they could be split into multiple PRs?

aaronjorbin committed:
use prettier to format CSS and JS

npx prettier@npm:wp-prettier@latest --write src/js/_enqueues/admin/site-icon.js

npx prettier@npm:wp-prettier@latest --write src/wp-admin/css/site-icon.css

aaronjorbin commented:
I've gone back and forth on this for the last few hours. On one hand, the new design more closely resembles the block editor and the current browser UI is outdated. On the other, the new design does not feel ready as it came in very late in the release cycle and as James notes, it hasn't considered edge cases. The improvements that are reflected in the [screenshots](https://github.com/WordPress/wordpress-develop/pull/6109#issuecomment-1959001331) seem to resolve the known issues, but I'm unsure. I also worry about integrating in this in three different places and then doing further interaction.

As I've mentioned multiple times, I think what's going to provide the best experience for the users of WordPress is an experience that is consistent across the different UIs. Iterating on the UI in the three different places is cumbersome at best and bug-inducing at worst.

Unless there is an objection, I think the path forward is:
1. Rip out the new design and restore the battle-tested UI that is already in place in the customizer.
2. Revisit the UI in 6.6. With any major desired UI changes coming well before beta 1.
3. Port the alt text and description improvements to the customizer in a separate ticket. I would consider this a bug fix for a feature in 6.5 even though that UI isn't directly being touched(@swissspidy, would you agree?)

RE: Prettier. That command doesn't format correctly. I used 
`npx prettier@npm:wp-prettier@latest --write src/js/_enqueues/admin/site-icon.js` and `npx prettier@npm:wp-prettier@latest --write src/wp-admin/css/site-icon.css`.  Thanks!

swissspidy commented:
Thanks for the update & context @aaronjorbin! That sounds like a sensible path forward to me.

I see you just opened #6165 in the meantime. Would you like the group to review that one now? Just noting it's still in draft.

aaronjorbin commented:
I opened that PR so that it's ready if this path forward is desired without further clouding this one. The only differences are 

1: I fixed the jshint issues that prettier caused 🤦 
2: https://github.com/WordPress/wordpress-develop/pull/6165/commits/984a20ca6ddae6e945838b207a22da0048fe599a is the commit that restores the old UI. Reviews of that are welcome; it's more red than green.
